### ISSUE TITLE
Why should we scale learning rate to num cores?

### ISSUE THREAD

**Issue by DayuanJiang** [pytorch/xla#3079]
https://github.com/pytorch/xla/blob/bc4aee9b854e8325fdbcccf514f0342263a0c8fe/test/test_profile_mp_mnist.py#L121

In the test scripts, the learning rate is scaled to num cores. Why should we do that?

And if I use a scheduler like [transformers.get_cosine_schedule_with_warmup](https://huggingface.co/transformers/main_classes/optimizer_schedules.html#transformers.get_cosine_schedule_with_warmup) in the training, should I also scale the training step to num cores like learning rate dose.  ( In another script, training steps is scaled but not learning rate)

https://github.com/pytorch/xla/blob/bc4aee9b854e8325fdbcccf514f0342263a0c8fe/test/test_train_mp_imagenet.py#L197
```python
  optimizer = optim.SGD(
      model.parameters(),
      lr=FLAGS.lr,
      momentum=FLAGS.momentum,
      weight_decay=1e-4)
  num_training_steps_per_epoch = train_dataset_len // (
      FLAGS.batch_size * xm.xrt_world_size())
````

**Comment by juneoh** [comment]
In [Accurate, Large Minibatch SGD: Training ImageNet in 1 Hour (2017)](https://arxiv.org/abs/1706.02677), adjusting the learning rate as a linear function of the batch size was introduced as an approach to help optimization in distributed, synchronous SGD. That should be the idea — the batch size that you pass to DataLoader is the batch size *per core*, so with the global batch size of `batch_size * xm.xrt_world_size()`, it may also make sense to scale the learning rate accordingly as `lr * xm.xrt_world_size()`.

However, I think manually scaling the learning rate as in `test_profile_mp_mnist.py` is no longer necessary, as [we already adjust the scale when reducing the gradients](https://github.com/pytorch/xla/blob/bc4aee9b854e8325fdbcccf514f0342263a0c8fe/torch_xla/core/xla_model.py#L801) during `xm.optimizer_step()`. Let me check with the team and fix that.

**Comment by rwightman** [comment]
@juneoh @DayuanJiang it's standard practice as per juneoh's comment to scale the lr linearly for SGD with the global batch size (and global batch size here increases with the number of processes). However it's more common to see it (in papers and practice) scaled per some fixed batch size like 256, where you'd specify lr per batch 256 and multiply that by global batch size which works better regardless of whether you're changing to batch size in a distributed or non-distributed setting.

It should also pointed out if you use a different optimizer (especially adaptive ones) like Adam, Lamb, etc it's usually better to scale with the sqrt of batch size ratio, not linearly.
